bottlerocket/agents/src/bin/eks-resource-agent/eks_provider.rs

```diff
@@ -239,10 +251,23 @@ struct ManagedNodeGroup {
     min_size: i32,
     /// The maximum number of nodes in the managed node group.
     max_size: i32,
-    // The desired number of nodes in the managed node group.
+    /// The desired number of nodes in the managed node group.
     desired_capacity: i32,
 }
 
+fn zone_to_exclude(exclude: String) -> Vec<AvailabilityZones> {
```

Comment:
This feels like it should be swapped out for a function using the EC2 client to get the available availability zones for that account in that region. 

nit: the name is not really helpful. You have named it what you are using it for, not what it does. This function returns the availability zones in us-west-2. If we must keep this function (which my above comment asks not to), then it should be named something more specific to its function, not what you want to use it for like `all_us_west_2_zones()` or something like that.